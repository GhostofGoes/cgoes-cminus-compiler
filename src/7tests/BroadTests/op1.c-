main()
{
    outputb(111<=222);
    outputb(222<=111);
    outputb(111<=111);
    outnl();

    outputb(111<222);
    outputb(222<111);
    outputb(111<111);
    outnl();

    outputb(111>222);
    outputb(222>111);
    outputb(111>111);
    outnl();

    outputb(111>=222);
    outputb(222>=111);
    outputb(111>=111);
    outnl();

    outputb(111==222);
    outputb(222==111);
    outputb(111==111);
    outnl();

    outputb(111!=222);
    outputb(222!=111);
    outputb(111!=111);
    outnl();

    outputb(111!=222);
    outputb(222!=111);
    outputb(111!=111);
    outnl();

    outputb(false  &  false);
    outputb(false  &  true);
    outputb(true  &  false);
    outputb(true  &  true);
    outnl();

    outputb(false  |  false);
    outputb(false  |  true);
    outputb(true  |  false);
    outputb(true  |  true);
    outnl();

    outputb( ! false);
    outputb( ! true);
    outnl();

    output(333+444);
    output(333-444);
    output(333*444);
    output(333/444);
    output(3333%444);
    output(-444);

    return;
}
