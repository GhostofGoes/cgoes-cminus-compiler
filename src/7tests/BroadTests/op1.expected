Number of warnings: 0
Number of errors: 0
------------------------------------

Memory Configuration: Data Addresses: 0-9999  Instruction Addresses: 0-9999
Loading file: op1.tm
T F T 
T F F 
F T F 
F T T 
F F T 
T T F 
T T F 
F F F T 
F T T T 
T F 
777 -111 147852 0 225 -444 Status: Halted
Bye.
